ppstructure/predict_system.py

```diff
@@ -116,6 +117,20 @@ def __call__(self, img, return_ocr_result_in_table=False, img_idx=0):
             else:
                 h, w = ori_im.shape[:2]
                 layout_res = [dict(bbox=None, label='table')]
+
+            # As reported in issues such as #10270 and #11665, the old
+            # implementation, which recognizes texts from the layout regions,
+            # has problems with OCR recognition accuracy.
+            #
+            # To enhance the OCR recognition accuracy, we implement a patch fix
+            # that first detect all text regions by using the text_detector
+            # and then recognize the texts from the text regions (intersecting
+            # with the layout regions) by using the text_recognizer.
```

Comment:
I prefer to use `text_system` to first detect and recognize all text information, and then filter out relevant text based on the layout information.

Reply:
I'll give this a try.

Reply:
One more question here, the following recover part has not been changed:

https://github.com/PaddlePaddle/PaddleOCR/blob/c82dd6406eecb3c0840b26b055b436a883128307/ppstructure/predict_system.py#L139-L143

Should we use the same mechanism for this piece of logic?

Reply:
It would be advisable to update all the code in this section to match the current logic as well.